include/seqan3/io/alignment_file/format_bam.hpp

```diff
@@ -0,0 +1,583 @@
+     *
+     * The function throws a seqan3::format_error if any unknown tag type was encountered. It will also fail if the
+     * format is not in a correct state (e.g. required fields are not given), but throwing might occur downstream of
+     * the actual error.
+     */
+    template <typename stream_view_type>
+    void read_one_tag(stream_view_type && stream_view, sam_tag_dictionary & target)
+    {
+        /* Every BA< tag has the format "[TAG][TYPE_ID][VALUE]", where TAG is a two letter
+           name tag which is converted to a unique integer identifier and TYPE_ID is one character in [A,i,Z,H,B,f]
+           describing the type for the upcoming VALUES. If TYPE_ID=='B' it signals an array of
+           VALUE's and the inner value type is identified by the next character, one of [cCsSiIf], followed
+           by the length (int32_t) of the array, followed by the values.
+        */
+        uint16_t tag = static_cast<uint16_t>(*std::ranges::begin(stream_view)) << 8;
+        std::ranges::next(std::ranges::begin(stream_view)); // skip char read before
+        tag += static_cast<uint16_t>(*std::ranges::begin(stream_view));
+        std::ranges::next(std::ranges::begin(stream_view)); // skip char read before
+        char type_id = static_cast<char>(*std::ranges::begin(stream_view));
+        std::ranges::next(std::ranges::begin(stream_view)); // skip char read before
+
+        switch (type_id)
+        {
+            case 'A' : // char
+            {
+                target[tag] = static_cast<char>(*std::ranges::begin(stream_view));
+                std::ranges::next(std::ranges::begin(stream_view)); // skip char that has been read
+                break;
+            }
+            // all integer sizes are possible
+            case 'c' : // int8_t
+            {
+                int8_t tmp;
+                read_field(stream_view, tmp);
+                target[tag] = static_cast<int32_t &&>(tmp); // readable sam format only allows int32_t
+                break;
+            }
+            case 'C' : // uint8_t
+            {
+                uint8_t tmp;
+                read_field(stream_view, tmp);
+                target[tag] = static_cast<int32_t &&>(tmp); // readable sam format only allows int32_t
+                break;
+            }
+            case 's' : // int16_t
+            {
+                int16_t tmp;
+                read_field(stream_view, tmp);
+                target[tag] = static_cast<int32_t &&>(tmp); // readable sam format only allows int32_t
+                break;
+            }
+            case 'S' : // uint16_t
+            {
+                uint16_t tmp;
+                read_field(stream_view, tmp);
+                target[tag] = static_cast<int32_t &&>(tmp); // readable sam format only allows int32_t
+                break;
+            }
+            case 'i' : // int32_t
+            {
+                int32_t tmp;
+                read_field(stream_view, tmp);
```

Comment:
First because the key for `target` (sam dict) is `tag` and not `tmp` :p and second because the value type of the sam dict is a std::variant, if I pass `target[new_key]` that the value will be default initialized to the first alternative which is not int32_t. and then the read_field will read a wrong number of bytes

Reply:
yes I overlooked tmp/tag.